src/context/AppContext.tsx

```diff
@@ -10,10 +10,26 @@ type Props = {
 
 export const AppContextProvider: React.FC<Props> = ({ children }) => {
   const [cart, setCart] = useState<Phone[]>([]);
+  const [addedIds, setAddedIds] = useState<string[]>([]);
+
+  const addToCart = (product: Phone) => {
+    setCart([...cart, product]);
+    setAddedIds([...addedIds, product.id]);
+  };
```

Comment:
We can improve here to save only required fields in cart
``id`` ``name`` ``priceRegular`` or ``priceDiscount`` depends which one is smaller, ``images[0]`` as image and ``quantity``
For types you can use [Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) to pick needed keys